restricted_site_access.php

```diff
@@ -87,6 +90,35 @@ public static function ajax_notice_dismiss() {
 		wp_send_json_success();
 	}
 
+	/**
+	 * Keeps log of network-wide RSA disable action
+	 *
+	 * @return void
+	 */
+	public static function ajax_network_disable() {
+		if ( ! check_ajax_referer( 'rsa_admin_nonce', 'nonce', false ) ) {
+			wp_send_json_error();
+			exit;
+		}
+
+		if ( ! RSA_IS_NETWORK ) {
+			wp_send_json_error();
+			exit;
+		}
+
```

Comment:
Add a capability check, only superadmins (?) should be able to disable the plugin?

Reply:
This method is intended for the unique purpose to log network-wide disabling.

If this action was able to run with a valid nonce, and a non-super admin did fire it, then something is broken on the core and worthy to investigate.

Reply:
removed